src/live-watch/live-watch.ts

```diff
@@ -0,0 +1,172 @@
+        const onWillStopSession = tracker.onWillStopSession(session => {
+            this.activeSession = session;
+            this.refresh();
+        });
+        this._context.subscriptions.push(onDidChangeActiveDebugSession);
+        this._context.subscriptions.push(onWillStartSession);
+        this._context.subscriptions.push(onStopped);
+        this._context.subscriptions.push(onStackTrace);
+        this._context.subscriptions.push(onWillStopSession);
+    }
+
+    handleOnWillStartSession(session: GDBTargetDebugSession): void {
+        session.refreshTimer.onRefresh(async (refreshSession) => {
+            this.activeSession = refreshSession;
+            this.refresh();
+        });
+    }
+    addVSCodeCommands() {
+        this._context.subscriptions.push(
+            vscode.window.registerTreeDataProvider('cmsis-debugger-view', this)
+        );
+        this._context.subscriptions.push(
+            vscode.commands.registerCommand('cmsis-debugger-view.add', async () => {
+                const expression = await vscode.window.showInputBox({ prompt: 'Expression' });
+                if (!expression) return;
+                await this.add(expression);
+            }),
+
+            vscode.commands.registerCommand('cmsis-debugger-view.clear', async () => {
+                const confirm = await vscode.window.showWarningMessage('Clear all expressions?', { modal: true }, 'Yes');
+                if (confirm === 'Yes') await this.clear();
+            }),
+
+            vscode.commands.registerCommand('cmsis-debugger-view.delete', async (node) => {
+                if (!node) return;
+                await this.delete(node);
+            }),
+
+            vscode.commands.registerCommand('cmsis-debugger-view.refresh', () => this.refresh()),
+
+            vscode.commands.registerCommand('cmsis-debugger-view.rename', async (node) => {
+                if (!node) return;
+                const expression = await vscode.window.showInputBox({ prompt: 'Expression', value: node.expression });
+                if (!expression) return;
+                await this.rename(node, expression);
+            })
+        );
+    }
+    async evaluate(expression: string): Promise<string> {
+        if (!this.activeSession) {
+            return 'No active session';
+        }
+        const result = await this.activeSession.evaluateGlobalExpression(expression);
+        return result;
+    }
+
+    async add(expression: string, parent?: LiveWatchNode) {
+        // Create a new node with a unique ID and evaluate its value
+        const newNode: LiveWatchNode = {
+            id: this.roots.length + 1,
+            expression,
+            value : await this.evaluate(expression),
+            parent: parent ?? undefined
+        };
+        if (!parent) {
+            this.roots.push(newNode);
+        } else {
+            parent.children?.push(newNode);
+        }
+        await this.save();
+        this.refresh();
```

Comment:
Probably return the promise from `this.refresh()`? Otherwise you may get an "outer" promise which resolves before the `this.refresh()` promise resolves. Or await `this.refresh()` here. Same for other `this.refresh()`  calls in below methods.